src/Simulation/Simulators.Type1/FindNuspecReferences.ps1

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+########################################
+# When creating a package with dotnet pack, nuget changes every ProjectReference to be itself
+# a PackageReference (without checking if that project has a corresponding package).
+# This is problematic because we currently don't want to create a package for every dll.
+#
+# On the other hand, when creating a package using nuget pack, nuget does not
+# identify PackageReferences defined in the csproj, so all the dependencies
+# are not listed and the package doesn't work.
+#
+# We don't want to hardcode the list of dependencies on the .nuspec, as they can
+# quickly become out-of-sync.
+# This script will find the PackageReferences recursively on the simulation projects and add them
+# to the nuspec, so we can then create the package using nuget pack with the corresponding
+# dependencies listed.
+#
+# nuget is tracking this problem at: https://github.com/NuGet/Home/issues/4491
+########################################
+
+$target = Join-Path $PSScriptRoot "Microsoft.Quantum.Simulators.Type1.nuspec"
+
+if (Test-Path $target) { 
+    Write-Host "$target exists. Skipping generating new one."
+    exit
+ }
+
+
+# Start with the nuspec template
+$nuspec = [xml](Get-Content (Join-Path $PSScriptRoot "Microsoft.Quantum.Simulators.Type1.nuspec.template"))
+$dep = $nuspec.CreateElement('dependencies', $nuspec.package.metadata.NamespaceURI)
+
+function Add-PackageReferenceIfNew($ref)
```

Comment:
Agreed (in fact, PowerShell is one of the few languages where it's not just a style difference but it can have functional implications as well, but that's a soap box for another time ;)). I updated this in the Type1 and Type2 FindNuspecReferences.ps1 files, but not in the existing ones that I haven't introduced or edited, for the sake of keeping the feature branch focused on decomposition related work.